### PR TITLE
[WIP] New planning tests

### PR DESCRIPTION
**Pull request by otsuman**
{{pytest: cfme/tests/optimize/test_planning.py }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 9ef3ec4171c067cf4ef4f31e78b4d7a39a9b30e8:

`cfme/optimize/planning.py`:
- :bangbang: Line 93:1: **E302** *expected 2 blank lines, found 1*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/optimize/planning.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
+from navmazing import NavigateToAttribute
+from widgetastic.widget import View
+from widgetastic_manageiq import (Button, Table, Checkbox, Input, LineChart,
```
[the remark this answers is not in the document]
Comment (rlbabyuk) [reply cfme/optimize/planning.py]: @psav need to clarify here, as previously I was requested to move widgetastic_manageiq to external libs, and currently, we have multiple places the same usage as here. And the correct way 
to import it in local imports, right?

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by mshriver** [comment]
Commits need to be signed with an approved key to run in PRT

File: cfme/optimize/planning.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+from widgetastic_patternfly import Tab
+
+from cfme.base.login import BaseLoggedInPage
+from cfme.modeling.base import BaseCollection
+from cfme.utils.appliance.implementations.ui import navigator, CFMENavigateStep
+from widgetastic_manageiq import (BootstrapSelect, Button, Checkbox, Input, LineChart,
+                                  SummaryFormItem, Table)
+
+
+class PlanningFilter(BaseLoggedInPage):
```
Comment (mshriver) [inline cfme/optimize/planning.py]: Please use a name like `PlanningFilterView` or `PlanningFilterForm` so as not to confuse the class with an object modeling a filter itself. (Where I would expect attributes representing a PlanningFilter

File: cfme/optimize/planning.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+    @property
+    def is_displayed(self):
+        return (
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == ['Optimize', 'Planning'])
+
+
+class PlanningCollection(BaseCollection):
+    pass
```
Comment (mshriver) [inline cfme/optimize/planning.py]: Entity class for the collection? A collection isn't much use if it doesn't have an `ENTITY` to go with it.

File: entry_points.txt [hunk 4]
```diff
@@ -81,6 +81,7 @@ orchestration_templates = cfme.services.catalogs.orchestration_template:Orchestr
 physical_providers = cfme.physical.provider:PhysicalProviderCollection
 physical_servers = cfme.physical.physical_server:PhysicalServerCollection
 physical_switches = cfme.physical.physical_switch:PhysicalSwitchCollection
+planning = cfme.optimize.planning:PlanningCollection
```
Comment (mshriver) [inline entry_points.txt]: Collections should be plural names, and a bit more detailed. Consider `optimize_plans`

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by mshriver** [comment]
Per this BZ, this UI component is being removed in more recent versions of MIQ.

I'm going to close this PR, as its relevance for current automation efforts has diminished.

https://bugzilla.redhat.com/show_bug.cgi?id=1672934

**Pull request closed without merging**